README.md

```diff
@@ -1,57 +1,297 @@
+  provider = MsRestAzure::ApplicationTokenProvider.new(
+      ENV['AZURE_TENANT_ID'],
+      ENV['AZURE_CLIENT_ID'],
+      ENV['AZURE_CLIENT_SECRET'])
+  credentials = MsRest::TokenCredentials.new(provider)
+  resource_client = Azure::ARM::Resources::ResourceManagementClient.new(credentials)
+  resource_client.subscription_id = subscription_id
+  network_client = Azure::ARM::Network::NetworkManagementClient.new(credentials)
+  network_client.subscription_id = subscription_id
+  storage_client = Azure::ARM::Storage::StorageManagementClient.new(credentials)
+  storage_client.subscription_id = subscription_id
+  compute_client = Azure::ARM::Compute::ComputeManagementClient.new(credentials)
+  compute_client.subscription_id = subscription_id
 
-(Any additional resources or related projects)
+  #
+  # Managing resource groups
+  #
+  resource_group_params = Azure::ARM::Resources::Models::ResourceGroup.new.tap do |rg|
+    rg.location = WEST_US
+  end
 
-- Link to supporting information
-- Link to similar sample
-- ...
+  # Create Resource group
+  puts 'Create Resource Group'
+  print_group resource_client.resource_groups.create_or_update(GROUP_NAME, resource_group_params)
+
+  postfix = rand(1000)
+  storage_account_name = "rubystor#{postfix}"
+  puts "Creating a premium storage account with encryption off named #{storage_account_name} in resource group #{GROUP_NAME}"
+  storage_create_params = StorageModels::StorageAccountCreateParameters.new.tap do |account|
+    account.location = WEST_US
+    account.sku = StorageModels::Sku.new.tap do |sku|
+      sku.name = StorageModels::SkuName::PremiumLRS
+      sku.tier = StorageModels::SkuTier::Premium
+    end
+    account.kind = StorageModels::Kind::Storage
+    account.encryption = StorageModels::Encryption.new.tap do |encrypt|
+      encrypt.services = StorageModels::EncryptionServices.new.tap do |services|
+        services.blob = StorageModels::EncryptionService.new.tap do |service|
+          service.enabled = false
+        end
+      end
+    end
+  end
+  print_item storage_account = storage_client.storage_accounts.create(GROUP_NAME, storage_account_name, storage_create_params)
+```
+
+<a id="vnet"></a>
+### Create a virtual network
+
+```ruby
+puts 'Creating a virtual network for the VM'
+vnet_create_params = NetworkModels::VirtualNetwork.new.tap do |vnet|
+  vnet.location = WEST_US
+  vnet.address_space = NetworkModels::AddressSpace.new.tap do |addr_space|
+    addr_space.address_prefixes = ['10.0.0.0/16']
+  end
+  vnet.dhcp_options = NetworkModels::DhcpOptions.new.tap do |dhcp|
+    dhcp.dns_servers = ['8.8.8.8']
+  end
+  vnet.subnets = [
+      NetworkModels::Subnet.new.tap do |subnet|
+        subnet.name = 'rubySampleSubnet'
+        subnet.address_prefix = '10.0.0.0/24'
+      end
+  ]
+end
+print_item vnet = network_client.virtual_networks.create_or_update(GROUP_NAME, 'sample-ruby-vnet', vnet_create_params)
+```
+
+<a id="ipaddress"></a>
+### Create a public IP address
```

Comment:
comment suggestion: We set up the VM by creating an IP address and its network interface,